MetaMask/metamask-extension
# Updating controller dependency

Updating controller

Manual Testing Instructions:
This is a bit tricky to describe because the expected behavior is somewhat complicated. 

### Phishing-detection flow:
What happens now is that when users navigate to any site, we check if the controller has fetched/updated the lists to check against in the last hour. If an update
- **has not happened in the past hour** we initiate a fetch to update the locally stored phishing lists, but we don’t wait for it to resolve before we proceed to check the current domain against the (potentially) stale list. This opens a small window where if the user happens to visit a domain that was added to the list since the last fetch as their first network activity after the wait interval has expired the user will not see the phishing warning page. Once the fetch to update the lists resolves the user will be warned upon any refreshes or new traffic to this domain.
- **has happened in the last hour** we do not make any requests to fetch updated lists.

We expect gaps in coverage for lists newly added to the domain between when things get updated in the phishing lists and when they can get caught: The gap will equal the amount of time between when the domain gets added to the phishing list and how soon after the controllers one hour refresh interval has completed that a user navigates to a new site plus the time that it takes for the fetch of updated phishing lists (if there are any updates) to resolve.

### Manual Testing
**This is a bit tricky to describe because the expected behavior is somewhat complicated.** 

1.  Start a build with the environment variable `IN_TEST=1` this will set the refresh interval to fetch updates to 5 seconds instead of an hour
2. Mock traffic to fetch the phishing lists like we do in the `phishing-detection.spec.js` e2e test
    - Mock this URL= `https://static.metafi.codefi.network/api/v1/lists/eth_phishing_detect_config.json'
'https://static.metafi.codefi.network/api/v1/lists/eth_phishing_detect_config.json`
    - To return something like:
```
 {
    version: 2,
    tolerance: 2,
    fuzzylist: [],
    whitelist: [],
    blacklist: [<add_test_domain_here>],
}
```
3. Navigate to any page so that the fetch will initialize
4. After 5 seconds, navigate to the domain that has been blacklisted
5. You should see the warning page. 

_I would encourage that we test other similar scenarios to understand the gaps in coverage and ensure what I describe above is accurate, but this is the base case._
